cmake/macros/SundialsAddNvectorBenchmark.cmake

```diff
@@ -0,0 +1,41 @@
+# ---------------------------------------------------------------
+# Programmer(s): Cody J. Balos @ LLNL
+# ---------------------------------------------------------------
+# SUNDIALS Copyright Start
+# Copyright (c) 2002-2024, Lawrence Livermore National Security
+# and Southern Methodist University.
+# All rights reserved.
+#
+# See the top-level LICENSE and NOTICE files for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+# SUNDIALS Copyright End
+# ---------------------------------------------------------------
+# CMake macro for adding nvector benchmark executables.
+# ---------------------------------------------------------------
+
+macro(sundials_add_nvector_benchmark NAME)
+
+  set(options)
+  set(singleValueArgs)
+  set(multiValueArgs SOURCES SUNDIALS_TARGETS LINK_LIBRARIES INSTALL_SUBDIR)
+
+  cmake_parse_arguments(arg "${options}" "${singleValueArgs}"
```

Comment:
To avoid any variable collisions, either make this a function or change `arg` to something unique e.g., `sundials_add_nvector_benchmark_arg`